app/controllers/profile.js

```diff
@@ -0,0 +1,70 @@
+import Controller from 'ember-controller';
+import service from 'ember-service/inject';
+import config from 'wnyc-web-client/config/environment';
+import fetch from 'fetch';
+import RSVP from 'rsvp';
+
+const FLASH_MESSAGES = {
+  email: 'Your email has been updated. Remember to use this new email the next time you log on.',
+  password: 'Your password has been updated.'
+};
+
+export default Controller.extend({
+  session: service(),
+  flashMessages: service(),
+  
+  authenticate(password) {
+    let email = this.get('model.email');
+    return this.get('session').authenticate('authenticator:nypr', email, password);
+  },
+  
+  changePassword(changeset) {
+    let old_password = changeset.get('currentPassword');
+    let new_password = changeset.get('newPassword');
+    return new RSVP.Promise((resolve, reject) => {
+      this.get('session').authorize('authorizer:nypr', (header, value) => {
+        fetch(`${config.wnycAuthAPI}/v1/password`, {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+            Authorization: value
+          },
+          body: JSON.stringify({old_password, new_password})
+        })
```

Comment:
refactor to match the rest of the branch in formatting and use `rejectUnsuccessfulResponse`